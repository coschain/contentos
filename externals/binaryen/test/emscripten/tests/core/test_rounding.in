#include <stdio.h>
#include <math.h>

int main() {
  printf("%.1f ", round(1.4));
  printf("%.1f ", round(1.6));
  printf("%.1f ", round(-1.4));
  printf("%.1f ", round(-1.6));

  printf("%.1f ", round(1.5));
  printf("%.1f ", round(2.5));
  printf("%.1f ", round(-1.5));
  printf("%.1f ", round(-2.5));

  printf("%ld ", lrint(1.4));
  printf("%ld ", lrint(1.6));
  printf("%ld ", lrint(-1.4));
  printf("%ld ", lrint(-1.6));

  printf("%ld ", lrint(1.5));
  printf("%ld ", lrint(2.5));
  printf("%ld ", lrint(-1.5));
  printf("%ld ", lrint(-2.5));

  printf("\n");

  double param, fractpart, intpart;

  param = 3.14159265;
  fractpart = modf (param , &intpart);
  printf ("%f = %f + %f \n", param, intpart, fractpart);

  param = -3.14159265;
  fractpart = modf (param , &intpart);
  printf ("%f = %f + %f \n", param, intpart, fractpart);

  printf("%.1f ", roundf(1.4));
  printf("%.1f ", roundf(1.6));
  printf("%.1f ", roundf(-1.4));
  printf("%.1f ", roundf(-1.6));

  printf("%.1f ", roundf(1.5));
  printf("%.1f ", roundf(2.5));
  printf("%.1f ", roundf(-1.5));
  printf("%.1f ", roundf(-2.5));

  return 0;
}
