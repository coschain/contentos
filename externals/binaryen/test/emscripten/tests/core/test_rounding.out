1.0 2.0 -1.0 -2.0 2.0 3.0 -2.0 -3.0 1 2 -1 -2 2 2 -2 -2 
3.141593 = 3.000000 + 0.141593 
-3.141593 = -3.000000 + -0.141593 
1.0 2.0 -1.0 -2.0 2.0 3.0 -2.0 -3.0 
